modules/alerts/main.tf

```diff
@@ -0,0 +1,59 @@
+resource "google_monitoring_alert_policy" "alert_policy" {
+  for_each = { for alert in var.policies : alert.display_name => alert }
+
+  display_name          = each.value["display_name"]
```

Comment:
it's not critical, but you could just do `each.value.display_name`, (the same for the other ones)